src/payment/strategies/amazon-pay/amazon-pay-payment-strategy.spec.ts

```diff
@@ -435,4 +440,91 @@ describe('AmazonPayPaymentStrategy', () => {
             expect(error).toBeInstanceOf(OrderFinalizationNotRequiredError);
         }
     });
+
+    describe('When 3ds is enabled', () => {
+        let payload: any;
+        let options: any;
+        let store3ds: CheckoutStore;
+        let strategy3ds: AmazonPayPaymentStrategy;
+        let amazonConfirmationFlow: AmazonPayConfirmationFlow;
+
+        beforeEach(async () => {
+            payload = getOrderRequestBody();
+            options = { methodId: paymentMethod.id };
+
+            store3ds = createCheckoutStore({
+                remoteCheckout: {data: {}}, paymentMethods: getPaymentMethodsStateWith3ds(),
+            });
+            strategy3ds = new AmazonPayPaymentStrategy(
+                store3ds,
+                orderActionCreator,
+                billingAddressActionCreator,
+                remoteCheckoutActionCreator,
+                scriptLoader
+            );
+
+            amazonConfirmationFlow = {
+                success: jest.fn(),
+                error: jest.fn(),
+            };
+
+            jest.spyOn(store3ds, 'dispatch').mockReturnValue(Promise.resolve());
+            jest.spyOn(store3ds.getState().remoteCheckout, 'getCheckout')
+                    .mockReturnValue({ referenceId: '511ed7ed-221c-418c-8286-f5102e49220b' });
+            await strategy3ds.initialize({ methodId: paymentMethod.id, amazon: { container: 'wallet' } });
+        });
+
+        it('redirect to confirmation flow when support 3ds', async () => {
+            // tslint:disable-next-line:no-non-null-assertion
+            hostWindow.OffAmazonPayments!.initConfirmationFlow = jest.fn((sellerId, referenceId, callback) => {
+                callback(amazonConfirmationFlow);
+            });
+
+            strategy3ds.execute(payload, options);
+
+            await new Promise(resolve => process.nextTick(resolve));
```

Comment:
Because `strategy3ds.execute` won't be resolved so I need this line to pass the test actually I based this test on some strategies tests that has a similar implementation